### PR TITLE
Implement forgot password

### PR DESCRIPTION
**Pull request by applenele**
reset password

### CODE REVIEW

File: src/CodeComb.vNextChina/Controllers/AccountController.cs [hunk 1]
```diff
@@ -132,6 +132,104 @@ public IActionResult RegisterDetail(string key)
...
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Forgot(string email)
+        {
+            // 判断该邮箱是否已经存在 不存在不能密码重置
+            if (!DB.Users.Any(x => x.Email == email))
+                return Prompt(x =>
+                {
+                    x.Title = "密码重置失败";
```
Comment (yukozh) [inline src/CodeComb.vNextChina/Controllers/AccountController.cs]: The title should be 找回密码失败

File: src/CodeComb.vNextChina/Controllers/AccountController.cs [hunk 2]
```diff
@@ -132,6 +132,104 @@ public IActionResult RegisterDetail(string key)
...
+            <head></head>
+            <body>
+            <p><a href=""{url}"">点击继续完成密码重置</a></p>
+            </body>
+            </html>");
+
+            return Prompt(x =>
+            {
+                x.Title = "请验证您的邮箱";
+                x.Details = $"我们向您的邮箱{email}中发送了一条包含验证链接的邮件，请通过邮件打开链接继续完成密码重置操作";
```
Comment (yukozh) [inline src/CodeComb.vNextChina/Controllers/AccountController.cs]: 找回密码

File: src/CodeComb.vNextChina/Controllers/AccountController.cs [hunk 3]
```diff
@@ -132,6 +132,104 @@ public IActionResult RegisterDetail(string key)
...
+        public IActionResult ForgotDetail(string key)
+        {
+            // 判断该邮箱是否已经存在 不存在不能密码重置
+            var email = Aes.Decrypt(key);
+            ViewBag.Key = key;
+            ViewBag.Email = email;
+            if (!DB.Users.Any(x => x.Email == email))
+                return Prompt(x =>
+                {
+                    x.Title = "密码重置失败";
```
Comment (yukozh) [inline src/CodeComb.vNextChina/Controllers/AccountController.cs]: 找回密码

File: src/CodeComb.vNextChina/Properties/launchSettings.json [hunk 4]
```diff
@@ -1,17 +1,18 @@
...
+      "environmentVariables": {
+        "ASPNET_ENV": "Development"
+      }
+    },
+    "web": {
+      "commandName": "web",
+      "environmentVariables": {
+        "ASPNET_ENV": "Development"
+      },
+      "sdkVersion": "dnx-clr-win-x64.1.0.0-rc2-16176"
```
Comment (yukozh) [inline src/CodeComb.vNextChina/Properties/launchSettings.json]: Do not change this file.

File: src/CodeComb.vNextChina/Views/Account/Forgot.cshtml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+﻿@{
+    ViewBag.Title = "新用户注册";
+}
+
+<div class="row">
+    <div class="col-md-8 col-sm-12 col-md-offset-2">
+        <div class="card card-single card-error">
+            <div class="error-block-inner">
+                <h3>密码重置</h3>
+                <p>在此我们需要验证您的邮箱合法性</p>
```
Comment (yukozh) [inline src/CodeComb.vNextChina/Views/Account/Forgot.cshtml:10]: 找回密码, ViewBag.Title

File: src/CodeComb.vNextChina/Views/Account/ForgotDetail.cshtml [hunk 6]
```diff
@@ -0,0 +1,30 @@
+﻿@{
+    ViewBag.Title = "新用户注册";
```
Comment (yukozh) [inline src/CodeComb.vNextChina/Views/Account/ForgotDetail.cshtml]: Change the title

File: src/CodeComb.vNextChina/Views/Account/ForgotDetail.cshtml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+<div class="row">
+    <div class="col-md-8 col-sm-12 col-md-offset-2">
+        <div class="card card-single card-error">
+            <div class="error-block-inner">
+                <h3>密码重置</h3>
+                <p>请完善您的个人资料</p>
```
Comment (yukozh) [inline src/CodeComb.vNextChina/Views/Account/ForgotDetail.cshtml]: this line should be 请设置新密码

File: src/CodeComb.vNextChina/Views/Account/ForgotDetail.cshtml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+                    <p>
+                        <span class="texthold">密码</span>
+                        <input type="password" class="textbox w-3" name="password" />
+                    </p>
+                    <p>
+                        <span class="texthold">密码重复</span>
+                        <input type="password" class="textbox w-3" name="confirm" />
+                    </p>
+                    <input type="hidden" name="key" value="@ViewBag.Key" />
+                    <input type="submit" value="注册" class="btn btn-default ripple home-link" />
```
Comment (yukozh) [inline src/CodeComb.vNextChina/Views/Account/ForgotDetail.cshtml]: Change the title of button

File: src/CodeComb.vNextChina/config.json [hunk 9]
```diff
@@ -2,7 +2,7 @@
   "Data": {
     "DefaultConnection": {
       "Mode": "SqlServer",
-      "ConnectionString": "Data source=localhost;database=vnextcn;uid=sa;password=Yuuko19931101"
```
Comment (yukozh) [inline src/CodeComb.vNextChina/config.json:5]: Do not change this

**Comment by yukozh** [comment]
Can you squash into one commit?

File: src/CodeComb.vNextChina/config.json (hunk 9 above)
Comment (applenele) [inline src/CodeComb.vNextChina/config.json:5]: 我本地没得数据库

File: src/CodeComb.vNextChina/Properties/launchSettings.json (hunk 4 above)
Comment (applenele) [inline src/CodeComb.vNextChina/Properties/launchSettings.json]: 这个文件 我不知道 自己变得

Comment (yukozh) [inline src/CodeComb.vNextChina/Properties/launchSettings.json]: To use the dev branch file overwrite this file.

**Comment by yukozh** [comment]
:shipit:

**Pull request merged**